src/MissionEditor/MissionEditor.qml

```diff
@@ -484,8 +501,7 @@ QGCView {
                             onClicked:  setCurrentItem(object.sequenceNumber)
 
                             onRemove: {
-                                itemDragger.clearItem()
-                                controller.removeMissionItem(object.sequenceNumber)
```

Comment:
I have tested. After changing the code from "controller.removeMissionItem(object.sequenceNumber)" to "controller.removeMissionItem(index)", deleting multi-mission of survey works well